dev/PushNotifications/PushNotificationsLongRunningTask/NotificationListener.cpp

```diff
@@ -28,3 +32,17 @@ STDMETHODIMP_(HRESULT __stdcall) NotificationListener::OnRawNotificationReceived
     return S_OK;
 }
 CATCH_RETURN()
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationListener::OnToastNotificationReceived(
+    ABI::Microsoft::Internal::ToastNotifications::INotificationProperties* notificationProperties,
+    ABI::Microsoft::Internal::ToastNotifications::INotificationTransientProperties* notificationTransientProperties) noexcept try
+{
+    auto lock = m_lock.lock_exclusive();
```

Comment:
1. The code isn't substantially modifying  the members, the map operation is just a query. This can be a `lock_shared`.
2. Please scope the lock to `appId` assignment. We don't want go all thru the UDK and hit wpncore (RPC call) while being in a lock.